src/components/PostDetails.tsx

```diff
@@ -1,106 +1,123 @@
 import React from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Comment } from '../types/Comment';
+import { Post } from '../types/Post';
+
+type Props = {
+  comments: Comment[];
+  currentPost: Post;
+  addComment: (a: Comment) => Promise<void>;
```

Comment:
The `addComment` function should not expect a `Promise<void>` return type since it does not return a promise. Adjust the type definition accordingly.